ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/AddParensToApplicationFix.fs

```diff
@@ -54,23 +55,39 @@ type AddParensToApplicationFix(error: NotAFunctionError) =
         | head :: tail -> createAppExpr factory (factory.CreateAppExpr(expr, head, true)) tail
         | [] -> expr
 
-    let findAppsWithoutParens prefixAppExpr =
-        let rec collectAppliedExprsRec (prefixAppExpr: IPrefixAppExpr) prefixAppDataAcc appliedExprsAcc =
-            let appExprFcsType = prefixAppExpr.ArgumentExpression.TryGetFcsType()
+    let getMaxArgsCount (expr: IFSharpExpression) =
+        let fcsType = expr.TryGetFcsType()
+
+        if isNotNull fcsType && fcsType.IsFunctionType then
+            List.length (FcsTypesUtil.getFunctionTypeArgs fcsType) - 1
 
-            let maxArgsCount =
-                if isNull appExprFcsType || not appExprFcsType.IsFunctionType then None else
+        else
+            match expr.IgnoreInnerParens() with
+            | :? ILambdaExpr as lambda ->
+                lambda.PatternsEnumerable.Count()
 
-                let args = FcsTypesUtil.getFunctionTypeArgs appExprFcsType
-                Some(args |> List.tail |> List.length)
+            | :? IReferenceExpr as ref ->
+                match ref.Reference.GetFSharpSymbol() with
+                | :? FSharpMemberOrFunctionOrValue as value when
```

Comment:
Would be great to use naming convention from the rest of the plugin and FCS and to call it `mfv`.

Reply:
Another problem with `value` is it has a particular meaning in the compiler that is even shown in `FSharpMemberOrFunctionOrValue` name. Are you sure that this mfv is not a member and not a function?